app/queries/keywords_query.rb

```diff
@@ -1,19 +1,58 @@
 # frozen_string_literal: true
 
 class KeywordsQuery
+  QUERY_PARAMS = [:url_pattern, :keyword_pattern, { link_types: [] }].freeze
+
+  attr_reader :keywords, :url_match_count
+
   def initialize(user)
+    @user = user
     @keywords = user.keywords.select(Keyword.column_names.excluding('html'))
   end
 
-  def call
+  def call(params = {})
+    @keyword_pattern = params[:keyword_pattern]
+    @url_pattern = params[:url_pattern]
+    @link_types = params[:link_types]
+
+    where_name
+
+    where_result_links
+
     order_by_name
+
+    @url_match_count = result_links_filtered.count
+
+    self
   end
 
   private
 
-  attr_reader :keywords
```

Comment:
Why are we not using attr_reader now?

Reply:
It moved from `private` to `public` attr_reader, as I need to return multiple values: The @keywords relation but also the url_match_count.

Reply:
Update: Following Junan solution, we now use public methods~